torchao/quantization/qat/api.py

```diff
@@ -214,15 +212,21 @@ def _qat_config_transform(
         # Swap FakeQuantizedEmbedding -> nn.Embedding
         # Then apply the base config's transform function to quantize the model
         assert step == QATStep.CONVERT, "unexpected step '%s' in QATConfig" % step
-        assert base_config is not None, "expected `base_config` in convert step"
+
+        is_fakequant = isinstance(module, (FakeQuantizedLinear, FakeQuantizedEmbedding))
+        assert base_config is not None or is_fakequant, (
+            "expected `base_config` in convert step or fake quantized `module`"
+        )
         if isinstance(module, FakeQuantizedLinear):
             module = module.to_linear()
         elif isinstance(module, FakeQuantizedEmbedding):
             module = module.to_embedding()
-        else:
-            # Unrelated module, ignore
-            return module
-        return _QUANTIZE_CONFIG_HANDLER[type(base_config)](module, base_config)
+
+        return (
+            module
+            if base_config is None or is_fakequant
+            else _QUANTIZE_CONFIG_HANDLER[type(base_config)](module, base_config)
+        )
```

Comment:
@andrewor14 I relaxed the constraints in the convert step so that `base_config=None` is valid. Is this okay?

Reply:
Actually I already did it here: https://github.com/pytorch/ao/pull/2694. You should be able to revert these changes after I land it

Reply:
Thanks! I left a comment on that PR since I run into an error about empty base_config and weight_config after rebasing.